### PR TITLE
[gatsby-plugin-now] Add new package w/support for Gatsby Redirects

### PR DESCRIPTION
**Pull request by lucleray**
This PR adds support for Gatsby Redirects + ZEIT Now, through a gatsby plugin `gatsby-plugin-now`.

The plugin generates a file `__now_routes_g4t5bY.json` that is then consumed by `@now/static-build` to create appropriate routes.

### CODE REVIEW

File: packages/now-static-build/src/index.ts
Comment (AndyBitz) [inline packages/now-static-build/src/index.ts]: Semicolons are missing

**Review by AndyBitz — CHANGES_REQUESTED** [review]
This creates a new plugin for Gatsby to create redirects specifically for Now, right? I thought there already is a plugin which we'll have to consume instead.

**Comment by lucleray** [comment]
> This creates a new plugin for Gatsby to create redirects specifically for Now, right? I thought there already is a plugin which we'll have to consume instead.

There is already a Gatsby *action* that we're consuming with this plugin: https://www.gatsbyjs.org/docs/actions/#createRedirect.

We're consuming redirects created with this action and creating appropriate now routes .

File: packages/gatsby-plugin-now/package.json
Comment (lucleray) [reply packages/gatsby-plugin-now/package.json]: Sure 👍

File: packages/now-static-build/src/index.ts
Comment (lucleray) [reply packages/now-static-build/src/index.ts:272]: The relevant code is existing code.

File: packages/gatsby-plugin-now/package.json
Comment (lucleray) [reply packages/gatsby-plugin-now/package.json]: I agree @styfle , I'll remove before I merge

File: packages/gatsby-plugin-now/gatsby-node.js
Comment (AndyBitz) [inline packages/gatsby-plugin-now/gatsby-node.js]: This file sometimes uses `'` and sometimes `"` for strings, I think it would be good to make it all equal

File: packages/gatsby-plugin-now/gatsby-node.js
Comment (AndyBitz) [inline packages/gatsby-plugin-now/gatsby-node.js]: Is `public` correct here? I think it will get deployed as a file when it gets in there, maybe it should just be in `program.directory`?

Comment (lucleray) [reply packages/gatsby-plugin-now/gatsby-node.js]: I chose to put in `public` because when you run `gatsby build` locally, the file will be generated but we don't want users to push it to their version control system.

To improve that: I can choose a name that's less likely to conflict and delete the file in `@now/static-build` to avoid deploying it.

Comment (AndyBitz) [reply packages/gatsby-plugin-now/gatsby-node.js]: Right. Deleting it in a different package seems a bit unintuitive though, but I can't think of something better.

Maybe we could add a timestamp to the file and then search for the file in `@now/static-build` and then delete it.

Comment (lucleray) [reply packages/gatsby-plugin-now/gatsby-node.js]: Why a timestamp @AndyBitz?

Comment (AndyBitz) [reply packages/gatsby-plugin-now/gatsby-node.js]: @lucleray We could use the sha sum as well, but then we had to read the file etc. I thought of something like this `__now_redirects.<timestamp>.json`.

Comment (AndyBitz) [reply packages/gatsby-plugin-now/gatsby-node.js]: Just to make sure it is not a fixed filename

**Review by AndyBitz — DISMISSED** [review]
Code looks good to me.

Does this package get published automatically? I'm not sure if there is more configuration required for this to work, maybe wait for @styfle to approve as well

Also, tests need to pass

File: packages/gatsby-plugin-now/index.js
Comment (styfle) [inline packages/gatsby-plugin-now/index.js:1]: Why create an index.js file at all?

File: .circleci/config.yml
Comment (styfle) [inline .circleci/config.yml]: Please rename `tests` to `test` to match the other packages.

File: packages/gatsby-plugin-now/package.json
Comment (styfle) [inline packages/gatsby-plugin-now/package.json]:
```suggestion
    "test-integration-once": "jest --env node --verbose --runInBand"
```

Can we keep it consistent with the other packages?

File: packages/gatsby-plugin-now/package.json
Comment (styfle) [inline packages/gatsby-plugin-now/package.json:26]: Is this jest config necessary?

File: packages/gatsby-plugin-now/readme.md
Comment (styfle) [inline packages/gatsby-plugin-now/readme.md]:
```suggestion
This plugin generates [Now Routes](https://zeit.co/docs/v2/advanced/routes) for [redirects](https://www.gatsbyjs.org/docs/actions/#createRedirect) you configured for to your Gatsby project.
```

Let use the Canonical URL which doesn't have the trailing slash

File: packages/gatsby-plugin-now/readme.md
Comment (styfle) [inline packages/gatsby-plugin-now/readme.md]:
```suggestion
1. Install the plugin:
```

File: packages/gatsby-plugin-now/package.json
Comment (styfle) [inline packages/gatsby-plugin-now/package.json]:
```suggestion
    "jest": "24.9.0"
```

Let's use exact versions for deps

File: packages/now-static-build/test/fixtures/10a-gatsby-redirects/gatsby-node.js
Comment (styfle) [inline packages/now-static-build/test/fixtures/10a-gatsby-redirects/gatsby-node.js:20]: Does status code 404 work? Perhaps this should be 307 instead?

File: packages/gatsby-plugin-now/index.js
Comment (lucleray) [reply packages/gatsby-plugin-now/index.js:1]: I followed [this guide](https://www.gatsbyjs.org/docs/files-gatsby-looks-for-in-a-plugin/#what-files-does-gatsby-look-for-in-a-plugin) which recommends it

File: packages/gatsby-plugin-now/package.json
Comment (lucleray) [reply packages/gatsby-plugin-now/package.json:26]: Yes, it is necessary to avoid `jest` running tests that are auto generated by gatsby in the fixtures folder

File: packages/gatsby-plugin-now/package.json
Comment (lucleray) [reply packages/gatsby-plugin-now/package.json]: `--runInBand` and `--env node` are not necessary.

```suggestion
    "test-integration-once": "jest --verbose"
```

File: packages/now-static-build/test/fixtures/10a-gatsby-redirects/gatsby-node.js
Comment (lucleray) [reply packages/now-static-build/test/fixtures/10a-gatsby-redirects/gatsby-node.js:20]: Yep it is how Gatsby shows example usage of `statusCode`.

**Review by styfle — APPROVED** [review]

**Pull request merged**
